common/CMakeLists.txt

```diff
@@ -23,3 +23,19 @@ if(${kv_db_sources} MATCHES ".*lmdb.*")
 	target_link_libraries(common_kv_db PUBLIC "${LMDB_LIB}")
   endif()
 endif()
+
+
+option(MENDER_LOG_BOOST "Use Boost as the underlying logger (default: Yes)" ON)
+
+add_library(common_log STATIC)
+target_include_directories(common_log PRIVATE ../)
+target_include_directories(common_log PUBLIC "${CMAKE_BINARY_DIR}/common")
+# Accept the global compiler flags
+target_link_libraries(common_log PUBLIC mender_compiler_flags)
+if (MENDER_LOG_BOOST)
+  find_package(Boost 1.73 REQUIRED COMPONENTS ALL)
+  target_sources(common_log PUBLIC log/platform/boost/boost_log.cpp)
+  target_link_libraries(common_log PUBLIC ${Boost_LIBRARIES})
+endif ()
```

Comment:
Hmm, yes. that is fine. The option was really added so that it can be changed out with an option. And initially I had two options, one for the boost log implementation, and one for the `plog` library. In which case straight finds are not the way we want to go I think

Reply:
I'd also go the way of detecting what's available. Potentially with some order of preference and options to enforce things.